keyboards/spacetime/keymaps/kyleterry/keymap.c

```diff
@@ -0,0 +1,168 @@
+    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,   KC_ESC,     KC_ENT,  KC_H,    KC_J,   KC_K,    KC_L,     KC_SCLN, KC_QUOT,
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,   NUM,        OS_LGUI, KC_N,    KC_M,   KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT,
+        KC_LALT,                        KC_V,    KC_SPC, LOWER,      RAISE,   KC_SPC,  OS_LGUI,                        KC_RCTL
+  ),
+
+  [_LOWER] = LAYOUT(
+    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LPRN,   KC_RPRN, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
+    _______, _______, _______, _______, _______, _______, KC_LBRC,   KC_RBRC, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
+    _______, _______, _______, _______, _______, _______, KC_LCBR,   KC_RCBR, _______, _______, _______, _______, TD(TD_MD), KC_MNXT,
+        _______,                        _______, _______, _______,   _______, _______, _______,                        _______
+  ),
+
+  [_RAISE] = LAYOUT(
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    _______,   _______, KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
+    _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,     _______, _______, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
+    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,    _______, _______, _______, _______, _______, KC_VOLD, KC_VOLU,
+        _______,                        _______, _______, _______,   _______, _______, _______,                        _______
+  ),
+
+  [_ADJUST] = LAYOUT(
+    _______, _______, KC_PGUP, _______, _______, _______, SQWER,     _______, _______, _______, KC_INS,  _______, KC_PSCR, _______,
+    _______, KC_HOME, KC_PGDN, KC_END,  _______, _______, SGAME,     _______, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______, _______, _______, _______, _______,
+        _______,                        _______, _______, _______,   _______, _______, _______,                        _______
+  ),
+
+  [_NUM] = LAYOUT(
+    _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______, KC_7,    KC_8,    KC_9,    KC_ASTR,
+    _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______, KC_4,    KC_5,    KC_6,    KC_MINS,
+    _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______, KC_1,    KC_2,    KC_3,    KC_PLUS,
+        _______,                        _______, _______, _______,   _______, _______, _______,                        KC_0
+  )
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+void matrix_init_user(void) {
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
+
+void tap_key(uint16_t keycode) {
```

Comment:
If you really need 16-bit keycodes (ie. shifted keycodes), this can be replaced with `tap_code16(kc)`.

Reply:
^ This. 

We have `tap_code` for 8 bit keycodes, and `tap_code16` for 16 bit (modded/shifted keycodes).

https://docs.qmk.fm/#/feature_macros?id=tap_codeltkcgt

Reply:
Awesome! This is good to know. There's a lot of keymaps that have functions like the one I had in the repo. Are these slowly being cleaned up?

Reply:
We're not removing them from existing boards/keymaps.  However, like this, when we do see them, we flag them. 

Also, very common functionality is slowly being ported into the core code, and documentation added for it.